plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskLists.php

```diff
@@ -404,25 +403,28 @@ function ( $all, $curr ) {
 	/**
 	 * Return number of setup tasks remaining
 	 *
-	 * @return number
+	 * This is not updated immediately when a task is completed, but rather when task is marked as complete in the database to reduce performance impact.
+	 *
+	 * @return int|null
 	 */
 	public static function setup_tasks_remaining() {
 		$setup_list = self::get_list( 'setup' );
 
-		if ( ! $setup_list || $setup_list->is_hidden() || $setup_list->has_previously_completed() || $setup_list->is_complete() ) {
+		if ( ! $setup_list || $setup_list->is_hidden() || $setup_list->has_previously_completed() ) {
 			return;
 		}
 
-		$remaining_tasks = array_values(
+		$viewable_tasks  = $setup_list->get_viewable_tasks();
+		$completed_tasks = get_option( Task::COMPLETED_OPTION, array() );
```

Comment:
Since this is coming from an option, I think we should check if it's an array first

Reply:
Good point! Updated the code to check if it's an array in a7aaf495cf8f1a12882523151c097c673b050f7f.